app/db/repository/authenticator_repository.rb

```diff
@@ -70,13 +80,21 @@ def load_authenticator(type:, account:, service_id:)
 
             args[variable.resource_id.split('/')[-1].underscore.to_sym] = variable.secret.value
           end
+          webservice.annotations.each do |annotation|
+            next unless @data_object.const_get(:OPTIONAL_ANNOTATIONS).include?(annotation.name)
+
+            args[annotation.name.sub('/', '_').to_sym] = annotation.value
+          end
         end
 
         begin
           if @pkce_support_enabled
             allowed_args = %i[account service_id] +
                           @data_object.const_get(:REQUIRED_VARIABLES) +
-                          @data_object.const_get(:OPTIONAL_VARIABLES)
+                          @data_object.const_get(:OPTIONAL_VARIABLES) +
+                          @data_object.const_get(:OPTIONAL_ANNOTATIONS).map do |arg|
+                            arg.sub('/', '_').to_sym
+                          end
```

Comment:
Using annotations for all configuration points might be a pleasant UX. In this case, should we disallow storing the `client_secret` as an annotation, and force it to be a variable? May not want to tie permissions over the client secret to permissions over the webservice.

Reply:
@john-odonnell & I talked about this in our implementation security review. We ended up down several rabbit holes of impacts on UX that indicate there's likely a deeper discussion to be had here that's more than just the 2 of us, including things like additional complexity in mixed config (knowing which takes precedence and determining which is active when trying to debug malfunctioning policy, for example) and not encouraging/allowing customers to put secrets in policy (like here, if the OIDC client-secret variable was instead a hardcoded value in an annotation.  

We also spent some time talking about how some config pieces come from different people (an Okta admin vs. a Conjur admin, for example) who might have different degrees of comfort/familiarity and how some of those values might actually be better off with a little more friction in changing, while others could use less friction. For example, maybe changing the provider URL for an OIDC provider should be a little harder than rotating a client secret for that provider. We may want to make it so that some changes can get multiple people involved in a change (like we make PRs need a review from someone else) to avoid a single rogue person causing problems.

There's a lot to discuss here and a broader group needed for participating in this discussion, so for now we went with making TTL a variable as well for consistency. We'd like to have that bigger discussion though and after that it might make sense ot revisit this piece too.